### DISCUSSION TITLE
About embedding key in netty

### DISCUSSION THREAD

**Discussion by alexchauhk** [netty/netty#14041] [Q&A] [answered]
Hi everyone! I may need some help for this.

There is a private key embedded in the source code which you can see them from [here](https://github.com/netty/netty/blob/343879818495a6179e397f32034d6122980232ef/handler/src/main/java/io/netty/handler/ssl/SslUtils.java#L117) . The key are being used twice in OpenSSL.java ([here](https://github.com/netty/netty/blob/343879818495a6179e397f32034d6122980232ef/handler/src/main/java/io/netty/handler/ssl/OpenSsl.java#L410) and [here](https://github.com/netty/netty/blob/343879818495a6179e397f32034d6122980232ef/handler/src/main/java/io/netty/handler/ssl/OpenSsl.java#L482)).

What are the usage of the key?  It doesn't look like used for running test. Grateful if anyone can give me some guidance!

**Comment by normanmaurer** [comment, marked as answer]
It's just used to test for some functionally while init our SSL code. Basically probing what kind of features are supported by the native impl

**Reply by alexchauhk** [reply]
Thank you @normanmaurer ! That's really helpful!
